### DISCUSSION TITLE
in jupyter notebooks, copilot doesn't see the code from other cells

### DISCUSSION THREAD

**Discussion by Timotheeee** [community/community#63296] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I've noticed that when working on a cell in a notebook (in vscode), copilot appears to be unable to see the code from other cells. This makes it nearly unusable because it has no clue what variables were defined in the cells above. I'm pretty sure it was capable of reading from other cells in the past. Is there a way to change it so that it can see all of the code?

**Comment by bala-srm** [comment]
same issue here. in the last week or so the quality of suggestions has plummetted. Even pre defined variables in Jupyter notebook code cells is not being suggested in the subsequent cells. Keeps suggesting junk comment lines instead of code. Very disappointed after having paid. I have switched to Codeium, which is free and gives super results and supports many languages including python and works on Jupyter notebook on browser as well as a plugin or as a browser extension and works even on colab and jupyter lab

**Comment by mikhailsirenko** [comment]
Same here. Maybe this can be tweaked in settings? While working with `.py` or `.tex` files all good though.

**Comment by NickTruesdale** [comment]
Same here. This really tanks the workflow in notebooks, which was really solid just a week or two ago :( 

Hopefully, since it was working before, it's relatively quick regression to fix?

**Comment by PFLigthart** [comment]
I have had the same problem.

**Comment by bash-j** [comment]
I downgraded to vscode version 1.80.2 and am not having the zero context issue anymore in notebooks. But now the jupyter kernels aren't working. No run button at the top, select kernel shows nothing. So upgrade again. Disabled a bunch of extensions and it's working in the new version.

**Comment by bala-srm** [comment]
Its fixed in version 1.105. pl update and check. From: bash-j ***@***.***>Date: Tuesday, 22 August 2023 at 5:14 AMTo: community/community ***@***.***>Cc: Balaji Sundararaman ***@***.***>, Comment ***@***.***>Subject: Re: [community/community] in jupyter notebooks, copilot doesn't see the code from other cells (Discussion #63296)I downgraded to vscode version 1.80.2 and am not having the zero context issue anymore in notebooks.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by Adminpyramid** [comment]
I just need to know the source codes in writing this page
